src/mongo/db/query/get_executor.cpp

```diff
@@ -1389,9 +1387,7 @@ StatusWith<unique_ptr<PlanExecutor>> getExecutorDistinct(OperationContext* txn,
     while (ii.more()) {
         const IndexDescriptor* desc = ii.next();
         IndexCatalogEntry* ice = ii.catalogEntry(desc);
-        // The distinct hack can work if any field is in the index but it's not always clear
-        // if it's a win unless it's the first field.
-        if (desc->keyPattern().firstElement().fieldName() == parsedDistinct->getKey()) {
+        if (desc->keyPattern().hasField(parsedDistinct->getKey())) {
```

Comment:
The special distinct optimization implemented by getExecutorDistinct() has two cases:

1) The case where there is no query predicate, which is handled on the block that opens on line 1441, and

2) the case where this is a query predicate, which is handled after the aforementioned block closes.

This change looks to me like it will work for case 2 but not for case 1. Specifically, line 1450 is still setting the distinct field to 0, even though the distinct field might not be the leading field. I actually think that we should continue to disallow the DISTINCT_SCAN fast path over a trailing field in the index when there is no query predicate, since it is not obvious that this is a performance win. My suggestion about how to implement this is to leave line 1450 alone, but instead modify getDistinctNodeIndex() to reject indices whose leading field is not equal to the distinct field.

Does this suggestion make sense?